Sources/PenguinPipeline/Pipeline/GeneratorPipelineIterator.swift

```diff
@@ -40,7 +42,9 @@ extension PipelineIterator {
   ///
   /// Note: if the function is expected to be expensive, it's often a good idea to call `.prefetch()`
   /// on the returned iterator.
-  public static func fromFunction<T>(_ typeHint: T.Type, _ function: @escaping () throws -> T?)
+  public static func fromFunction<T>(
+    _: Type<T>, _ function: @escaping () throws -> T?
```

Comment:
Why did you get rid of the `typeHint` name?

Reply:
Wasn't used in the documentation or code.  But maybe it should come back.  WDYT?  I guess  I'd probably gone for these usages over trying to avoid writing `()->T in` in the closure:

```
fromFunction { deducedReturnType() }
```

or 

```
fromFunction { ()->ReturnType in
  let sthg = something()
  return sthg 
}
```